core/src/main/java/org/apache/iceberg/TableProperties.java

```diff
@@ -106,6 +106,8 @@ private TableProperties() {}
   public static final String DEFAULT_WRITE_METRICS_MODE = "write.metadata.metrics.default";
   public static final String DEFAULT_WRITE_METRICS_MODE_DEFAULT = "truncate(16)";
 
+  public static final String METADATA_SPLIT_SIZE = "read.metadata.split-size";
```

Comment:
Can we move this up by the other split size property? Also, the other name is `read.split.target-size` so let's use the same prefix. How about changing this to `read.split.metadata-target-size`?